json/lex_number.mbt

```diff
@@ -12,136 +12,132 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-fn lex_decimal_integer(
-  ctx : ParseContext,
-  ~start : Int
-) -> Result[Double, ParseError] {
+fn lex_decimal_integer(ctx : ParseContext, ~start : Int) -> Double!ParseError {
   for ; ; {
     match read_char(ctx) {
-      Some('.') => return lex_decimal_point(ctx, ~start)
-      Some('e' | 'E') => return lex_decimal_exponent(ctx, ~start)
+      Some('.') => return lex_decimal_point(ctx, ~start)!
```

Comment:
`return  f(..)!` could be optimized cc @Yu-zh 
things maybe a little complicated (we need detect if there is an outer try)